uvicorn/main.py

```diff
@@ -360,6 +360,7 @@ def run(self, sockets=None, shutdown_servers=True):
         elif config.uds is not None:
             # Create a socket using UNIX domain socket.
             server = await loop.create_unix_server(create_protocol, path=config.uds)
+            os.chmod(config.uds, 0o666)
```

Comment:
I think we perhaps only do this if *we've* created the file.
If the UDS already exists, then we should stick with whatever existing permissions and ownership it has, right?
Perhaps this?...

```python
set_perms = not os.path.exists(config.uds)
server = await loop.create_unix_server(create_protocol, path=config.uds)
if set_perms:
    os.chmod(config.uds, 0o666)
```

Seem reasonable?

Reply:
> I think we perhaps only do this if we've created the file.

`create_unix_server` does it for us. [doc says](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.create_unix_server), that it always return Server object, or dies on exception.

> If the UDS already exists, then we should stick with whatever existing permissions and ownership it has, right?

We already do it.

Reply:
> If the UDS already exists, then we should stick with whatever existing permissions and ownership it has, right?

Pardon, ignore prevoius answer. We don't have to worry about permissions of existing files. If `create_unix_server` fails, it will throw exception.

Reply:
> We don't have to worry about permissions of existing files. If create_unix_server fails, it will throw exception.

I don't think it raises any exception if you're running it against an existing UDS.
(Eg. try running uvicorn --uds, then quitting it and running again)

Reply:
> I don't think it raises any exception if you're running it against an existing UDS.

It does not throw exception at least if user has permission to access to existing UDS. 
If existing UDS after successful execution of `create_unix_server` has some different permissions, it will change back to 0666 after `os.chmod(config.uds, 0o666)`

Reply:
I get it. You want to save UDS's permission if it runs on existing UDS. Editing...

Reply:
> I get it. You want to save UDS's permission if it runs on existing UDS. Editing...
 
Exactly, yes. Or rather: Just don't chmod it at all in that situation.

Reply:
Eg. the UDS has been created by some completely separate supervisor process - we don't want to touch the permissions with uvicorn in that case.